Islandora-CLAW/Crayfish
# Rewrite as symfony-flex

**GitHub Issue**: https://github.com/Islandora-CLAW/CLAW/issues/1224

Depends on the branch - https://github.com/Islandora-CLAW/Crayfish-Commons/pull/26

# What does this Pull Request do?

This PR recreates the Houdini microservice using Symfony/flex.

I will work on a PR for claw-playbook and ansible-role-crayfish

# What's new?

Changes to the symfony/flex framework.
To configure the firewall you do that in `config/packages/security.yaml`
To configure logging you do that in `config/packages/monolog.yaml`
To configure crayfish commons you do that in `config/packages/crayfish_commons.yaml`

* Does this change require documentation to be updated? yes
* Does this change add any new dependencies? yes, changes
* Does this change require any other modifications to be made to the repository (ie. Regeneration activity, etc.)? no
* Could this change impact execution of existing code? no

# How should this be tested?

Currently I was testing by:
* Standing up a normal claw-playbook.
* Add a binary image to the Islandora 8 and get the URL for Fedora.
* Check out this branch.
* `cd ./Houdini`
* `composer install`
* `php ./bin/console server:start`
    Get the host and port, then ensure your `crayfish_commons.yaml` has the correct settings, including the location to the example `syn-settings.xml`
* `curl -i -H"Accept: image/png" -H"Apix-Ldp-Resource: <fedora url>" -H"X-Islandora-Args: -resize 100x>" http://localhost:<symfony port>/convert` -> 401 Unauthorized
* `curl -i -H"Accept: image/png" -H"Apix-Ldp-Resource: <fedora url>" -H"X-Islandora-Args: -resize 100x>" -H"Authorization: Bearer bob" http://localhost:<symfony port>/convert` -> 403 Forbidden
* `curl -o test.png -H"Accept: image/png" -H"Apix-Ldp-Resource: <fedora url>" -H"X-Islandora-Args: -resize 100x>" -H"Authorization: Bearer islandora" http://localhost:<symfony port>/convert` -> view the resized image in test.png

# Interested parties
@Islandora-CLAW/committers
